### ISSUE TITLE
Parse from byte-vector results in compile error

### ISSUE THREAD

**Issue by std-any-emplace** [nlohmann/json#2602]
This is a test-case from a larger code-base:

```
TEST_CASE("parse from byte-vector", "[compile-ok-3.7.3][compile-fail-3.9.1]")
{
	auto b = [](const char a_char){ return static_cast<std::byte>(a_char); };
	auto vec = std::vector{b('{'), b('}')};
	REQUIRE(nlohmann::json::parse(vec).dump() == "{}");
}
```

This was working with version 3.7.3. After upgrade to version 3.9.1 this fails to compile. I guess it will also fail with 3.8.0 b/c the problem may be related to the work on the InputAdapterType.

I am not sure, if it was ever intended to work before, but it did.

#### Which version of the library did you use?

- [x] latest release version 3.9.1
- [x] other release - please state the version: 3.7.3
- [ ] the `develop` branch

**Comment by nlohmann** [comment]
This could have been fixed with #2550. Can you please try the `develop` branch?

**Comment by std-any-emplace** [comment]
Yes it does compile on develop. Sorry for the duplicate.
